### PR TITLE
Feature/expand architecture

### PR DESCRIPTION
**Pull request by ErichBSchulz**
Most (but not all) of this new content is from the wiki

this page still needs some love (see the comments) but hopefully a step forward!

### CODE REVIEW

File: docs/core/architecture.md [hunk 1]
```diff
@@ -38,33 +40,79 @@ Namespaces are designated with "\".
...
-BAO stands for business access object.  BAOs map to DAOs and extend them with
+BAO stands for business access object
+([example](https://github.com/civicrm/civicrm-core/blob/master/CRM/Event/BAO/Event.php)).
+BAOs map to DAOs and extend them with
 the business logic of CiviCRM.  A lot of the meat of CiviCRM is found in the
 BAOs, for example they have the code that creates follow up activities when an
 activity is created, or create activities and populating custom fields when a
 pledge is created.
 
+!!! note
```
Comment (eileenmcnaughton) [inline docs/core/architecture.md]: This is incorrect - they are really really inconsistent - I would probably argue we should move towards removing all the add functions & only having create

**Comment by seanmadsen** [comment]
Ok I'm going to go ahead and merge this now, even if it's a little premature, because we'd like to get a link for this page published so that we can reference it in the wiki (due to this page being an uncommon case of a wiki page with content to migrate but also *sub-pages*). (I can explain more if needed.) More edits can happen in the page once it's published.

**Comment by seanmadsen** [comment]
Thanks for the feedback @eileenmcnaughton !

**Comment by totten** [comment]
I guess this is a bit of tangent, but it'd pretty tempting to move the "API" section above the bao/dao/form stuff. Given all the unit-tests/metadata/bindings (php/cli/ajax), it's usually better for a developer to tap in at that layer -- and drill-down to the others necessary.

File: docs/core/architecture.md (hunk 1 above)
Comment (ErichBSchulz) [inline docs/core/architecture.md]: Thanks @eileenmcnaughton!

**Pull request merged**